### PR TITLE
Convert `VContentLink` story to MDX

### PR DESCRIPTION
**Pull request by sepehrrezaei**
## Fixes

Related to #584 by @dhruvkb 

## Description

This PR updates the story of VContentLink to use the composition API (MDX). Here is the screenshot of the converted VContentLink to MDX in the storybook: 
Screenshot 2023-03-29 at 14 14 23

## Testing Instructions

Run the Storybook from this PR to see that VContentLink is working properly in the storybook.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/src/components/VContentLink/meta/VContentLink.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      ],
+    }
+  },
+  methods: {
+    onSelected(mediaType) {
+      this.selected = mediaType
+    },
+  },
+})
+
```
Comment (obulat) [inline frontend/src/components/VContentLink/meta/VContentLink.stories.mdx:62]:
```suggestion
# VContentLink

```

The <h1> was missing in this story.

**Review by obulat — CHANGES_REQUESTED** [review]
The stories are great!

One change request here is to replace all instances of `VContentList` with `VContentLink`.

File: frontend/src/components/VContentLink/meta/VContentLink.stories.mdx (hunk 1 above)
Comment (sepehrrezaei) [reply frontend/src/components/VContentLink/meta/VContentLink.stories.mdx:62]: Sure, Thanks

File: frontend/src/components/VContentLink/meta/VContentLink.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+# VContentLink
+
+<Description of={VContentLink} />
+
+<ArgsTable of={VContentLink} />
+
+## Default VContentLink
```
Comment (dhruvkb) [inline frontend/src/components/VContentLink/meta/VContentLink.stories.mdx:69]: Nit: `VContentLink` is implied here.

```suggestion
## Default
```

File: frontend/src/components/VContentLink/meta/VContentLink.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
```
Comment (dhruvkb) [inline frontend/src/components/VContentLink/meta/VContentLink.stories.mdx:70]: Optional for this PR: a little description about each variant would be nice.

File: frontend/src/components/VContentLink/meta/VContentLink.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+      layout: "horizontal",
+    }}
+    argTypes={contentLinkArgTypes}
+  >
+    {VContentLinkHorizontalStory.bind({})}
+  </Story>
+</Canvas>
+
+## Mobile VContentLink
+
```
Comment (dhruvkb) [inline frontend/src/components/VContentLink/meta/VContentLink.stories.mdx:103]: Please include a note that this story should be viewed in the Canvas tab. I was a bit confused when the "mobile" story was same as the desktop one when seen in the Docs tab.

**Review by dhruvkb — CHANGES_REQUESTED** [review]
This is really good! Thank you @sepehrrezaei. I do have some minor suggestions though.

**Review by dhruvkb — APPROVED** [review]
Approving, shelving the recommended changes for future improvements.

**Pull request merged**